src/Raven.Server/Documents/Queries/Dynamic/DynamicQueryRunner.cs

```diff
@@ -62,7 +62,6 @@ public override async Task<DocumentQueryResult> ExecuteQuery(IndexQueryServerSid
             using (QueryRunner.MarkQueryAsRunning(index.Name, query, token))
             {
                 var queryResult = await index.Query(query, queryContext, token);
-                queryResult.RaftCommandIndex = result.Index;
```

Comment:
Now we are always adding this, right? How does it affect Orchestrator, IIRC there was a piece of code that was waiting for that command to be applied, but it was only executed when this was null.

Reply:
1. Yes, we always add it now.
2. We wait only if it's an map-reduce index

https://github.com/ravendb/ravendb/blob/af779cea08874f70f7867156abf57c2095b6783c/src/Raven.Server/Documents/Sharding/Queries/AbstractShardedQueryProcessor.cs#L454-L456

Reply:
Indeed that was problematic because it caused that we called `WaitForExecutionOnAllNodesAsync()` on every dynamic group by query. Fixed in https://github.com/ravendb/ravendb/pull/15844/commits/62d63c3b99d86541210ace94fdc65416dc7cc3ad